src/main/ts/ui/Buttons.ts

```diff
@@ -0,0 +1,118 @@
+import { Editor } from "tinymce";
+
+const register = (editor: Editor): void => {
+  const onAction = () => editor.execCommand('mceEnhancedList');
+
+  editor.ui.registry.addButton('shan-enhanced-lists', {
+    type: 'button',
+    icon: 'smiley',
+    tooltip: 'shan-enhanced-list',
+    onAction: onAction
+  });
+
+  editor.ui.registry.addSplitButton('enhanced-list-unordered-list', {
+    icon: 'unordered-list',
+    tooltip: 'Enhanced unordered list',
+    presets: 'listpreview',
+    columns: 3,
+    fetch: (callback) => {
+      const items: unknown[] = [
+        {
+          type: 'choiceitem',
+          icon: 'list-bull-default',
+          text: 'Disc',
+          value: 'disc'
+        },
+        {
+          type: 'choiceitem',
+          icon: 'list-bull-circle',
+          text: 'Circle',
+          value: 'circle'
+        },
+        {
+          type: 'choiceitem',
+          icon: 'list-bull-square',
+          text: 'Square',
+          value: 'square'
+        },
+      ];
+      callback(items);
+    },
+    onAction: () => {
+      editor.execCommand('InsertOrderedList');
+    },
+    onItemAction: (api, value) => {
+      editor.execCommand('InsertOrderedList', false, { 'list-style-type': value });
+    },
+  });
+
+  editor.ui.registry.addSplitButton('enhanced-list-ordered-list', {
+    icon: 'ordered-list',
+    tooltip: 'Enhanced ordered list',
+    presets: 'listpreview',
```

Comment:
Question: What does this `presets` do? What happens if you leave it out?

Reply:
`presets` defines the schema, to be created as a component? In this case, it creates a component with `div` tag and append the predefined class `tox-collection--toolbar-lg` to the div. So the styling 48px height and width gets applied to the div because it contains the class name `tox-collection--toolbar-lg`.